### ISSUE TITLE
Weird behaviour listing non-ascii characters

### ISSUE THREAD

**Issue by Gabarito007** [meganz/sdk#816]
I'm using the new version of MEGAcmd, 0.9.5.0.
Good to say about the effort of the developers to make it UTF-8 capable.
Despite that, it shows a weird behaviour.

After fetch the list of files at cloud, I noticed that, sometimes, few non-ASCII characters appears in a wrong way.

Example:
Cd100/Dicas/Códigos do f��rum 12 - LaTeX.txt

Note that at the same line we see two ways to display the same non-ASCII character.

The right way to display that line would be this:
Cd100/Dicas/Códigos do fórum 12 - LaTeX.txt

'ó' and '��' are the same thing, even though they are displayed differently.
At the same line
Hex:
C3 B3..........<->..........EF BF BD EF BF BD
UTF-8 code.

This behavior uses to occurs at other lines, randomly, with others non-ASCII characters.

Explaining more: around 80% of the list are displayed correctly. Few times they aren't.
And at the same line we find correct and wrong display.
Something is going well, but a few needs to fix.

That is not important for me, but I think it may should've known.

**Comment by Gabarito007** [comment]
As requested, I sent a list of files with abnormal characters display to support@mega.nz.

Thanks.

**Comment by Gabarito007** [comment]
About the file I sent via e-mail, wrong character displays not the same "correct" character.
I mean, this '��' becomes a kind of wildcard to show what it cannot display correctly.
So, about the file, below is a table to explay what I want to say:

Line......wrong shown......correct character
53...................��..........................ó
188.................��..........................ã
189.................��..........................í
190.................��..........................ê
191.................��..........................ã

**Comment by Gabarito007** [comment]
I understand that it is the worst error to fix: random error and no clue why appear or may not appear, depending on its will.
# 🙁

**Comment by polmr** [comment]
@Gabarito007 , I've been taking a look at your list. The thing is .. that's nothing we can use. We'd need to have one of those failing files. if possible received via attachment and via MEGA exported link so that we can try to reproduce the behavior shown at your list in case the issue is because those filenames have characters codified unexpectedly.

**Comment by Gabarito007** [comment]
Oh, my...
I got it now.
You want not the list, but the file itself.
Sending two files via MEGA link:
https://mega.nz/#!2MZGCarR!PskvMBxH5WfGAdScjDo6Tjp6aw7p185_Gx5j_A2rH1w
https://mega.nz/#!GZp0GRTB!NoqgeZqaQb5HzlelnaGfxm0ZlfLv8Ytt4mnJJZO-iBw

**Comment by polmr** [comment]
nah. Unable to reproduce I'm afraid. If you download and upload the file to a new location do you still get the wrong characters? Do you have any particular filesystem?

**Comment by Gabarito007** [comment]
Yes, I know, it's very weird.
Every time I fetch the list, there are others wrong characters, and that wrong before becomes correct ones.
I need to go offline now, but I'll come back after 4 hours and will make tests moving files to other folders at cloud.
I use NTFS file system on Windows 7 64bits.

**Comment by Gabarito007** [comment]
But I do confirm: more than 90% of the list are good UTF-8 characters.

**Comment by Gabarito007** [comment]
I moved some files among folders, downloaded and uploaded again, but result is the same: sometimes non-ascii characters are shown right, others aren't, ramdonly.
Mystery.
Good part: less than 4% has such error.

**Comment by polmr** [comment]
I've created a structure of files similar to the one you sent us, and still can't reproduce. Let's keep this issue open in case someone else experiences the same and see any relation.
